index.js

```diff
@@ -1,27 +1,30 @@
-const bodyParser = require("body-parser");
 const express = require("express");
 require("dotenv").config();
-const app = express();
-const mongoose = require("mongoose");
-const todoRoutes = require("./routes/todo.js");
+const scrapeController = require("./controllers/webscrapeController");
+const { getDateRange } = require("./models/dates");
 
-main()
-  .then((res) => console.log(res))
-  .catch((err) => console.log(err));
+const { startDate, endDate } = getDateRange();
+const mongoose = require("mongoose");
 
 async function main() {
-  await mongoose.connect(process.env.DB);
-}
+  try {
+    await mongoose.connect(process.env.DB);
+    console.log("Connected to the database");
+    
+    const app = express();
 
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
+    app.use("/api/todo", require("./routes/todo.js"));
 
-app.get("/", (req, res) => {
-  res.send("hello world");
-});
+    app.get("/getDataByDates", (req, res) => {
+      scrapeController.scrapeEspn(startDate, endDate, req, res);
```

Comment:
users should be able to pass an argument of end date. Right now const { startDate, endDate } = getDateRange(); 

Rethink this logic

Reply:
consider passing startdate as a param that is YYYYMMDD that way you don't have to get date range unless it's not provided